Lib/os.py

```diff
@@ -343,86 +343,96 @@ def walk(top, topdown=True, onerror=None, followlinks=False):
     return _walk(fspath(top), topdown, onerror, followlinks)
 
 def _walk(top, topdown, onerror, followlinks):
```

Comment:
I suggest we remove the `_walk` function and simply put everything under `walk` because the second one was only necessary because of sys.audit not making sense for recursive calls.